statmoments/_native.pyx

```diff
@@ -275,81 +282,86 @@ def calc_central_moments(raw, ave, n, lm, rm, i, j):
   # Horner representation generated with sympy
-  if lm + rm == 2:                             # E(X Y)
-    M_11 = raw[0,:,0,:]
+  if lm + rm == 2:  # E(X Y)
+    M_11 = raw[0, :, 0, :]
     m = M_11[i, j] * inv_n - ave_i * ave_j
-  elif lm + rm == 3:                           # E(XX Y)
-    M_11 = raw[0,:,0,:]  # 0
-    M_12 = raw[0,:,1,:]
-    m = ave_j*(2*n*ave_i*ave_j - 2*M_11[i, j]) \
-        - ave_i*M_11[j, j].diagonal() \
+  elif lm + rm == 3:  # E(XX Y)
+    M_11 = raw[0, :, 0, :]  # 0
+    M_12 = raw[0, :, 1, :]
+    m = ave_j * (2 * n * ave_i * ave_j - 2 * M_11[i, j]) \
+        - ave_i * M_11[j, j].diagonal() \
         + M_12[i, j]
     m *= inv_n
-  elif lm == 2 and rm == 2:                    # E(XX YY)
-    M_11 = raw[0,:,0,:]  # 0
-    M_12 = raw[0,:,1,:]
-    M_22 = raw[1,:,1,:]  # 1
-    m = ave_i*(ave_i*M_11[j, j].diagonal() - 2 * M_12[i, j]) \
-        + ave_j*(4*ave_i*M_11[i, j] + ave_j*(M_11[i, i] - 3*n*ave_i*ave_i) - 2*M_12[j, i]) \
+  elif lm == 2 and rm == 2:  # E(XX YY)
+    M_11 = raw[0, :, 0, :]  # 0
+    M_12 = raw[0, :, 1, :]
+    M_22 = raw[1, :, 1, :]  # 1
+    m = ave_i * (ave_i * M_11[j, j].diagonal() - 2 * M_12[i, j]) \
+        + ave_j * (4 * ave_i * M_11[i, j] + ave_j * (M_11[i, i] - 3 * n * ave_i * ave_i) - 2 * M_12[j, i]) \
         + M_22[i, j]
     m *= inv_n
-  elif lm == 4 and rm == 4:                    # E(XXXX YYYY)
-    M_11 = raw[0,:,0,:]  # 0
-    M_12 = raw[0,:,1,:]
-    M_13 = raw[0,:,2,:]
-    M_14 = raw[0,:,3,:]
-    M_22 = raw[1,:,1,:]  # 1
-    M_23 = raw[1,:,2,:]
-    M_24 = raw[1,:,3,:]
-    M_33 = raw[2,:,2,:]  # 2
-    M_34 = raw[2,:,3,:]
-    M_44 = raw[3,:,3,:]  # 3
+  elif lm == 4 and rm == 4:  # E(XXXX YYYY)
+    M_11 = raw[0, :, 0, :]  # 0
+    M_12 = raw[0, :, 1, :]
+    M_13 = raw[0, :, 2, :]
+    M_14 = raw[0, :, 3, :]
+    M_22 = raw[1, :, 1, :]  # 1
+    M_23 = raw[1, :, 2, :]
+    M_24 = raw[1, :, 3, :]
+    M_33 = raw[2, :, 2, :]  # 2
+    M_34 = raw[2, :, 3, :]
+    M_44 = raw[3, :, 3, :]  # 3
     m = ave_i * (-4 * M_34[i, j] + ave_i * (6 * M_24[i, j] + ave_i * (M_13[j, j].diagonal() * ave_i - 4 * M_14[i, j]))) \
         + ave_j * (-4 * M_34[j, i] \
-                     + ave_i * (16 * M_33[i, j] + ave_i * (-24 * M_23[i, j] + ave_i * (-4 * M_12[j, j].diagonal() * ave_i + 16 * M_13[i, j]))) \
-                     + ave_j * (6 * M_24[j, i] \
-                                  + ave_i * (-24 * M_23[j, i] + ave_i * (36 * M_22[i, j] + ave_i * ((6 * ave_i) * M_11[j, j].diagonal() - 24 * M_12[i, j]))) \
-                                  + ave_j * (-4 * M_14[j, i] \
-                                               + ave_i * (16 * M_13[j, i] + ave_i * (16 * ave_i * M_11[i, j] - 24 * M_12[j, i])) \
-                                               + ave_j * (M_13[i, i] \
-                                                            + ave_i * (-4 * M_12[i, i] + ave_i * (6 * M_11[i, i] - 7* n * ave_i ** 2)))))) \
+                   + ave_i * (16 * M_33[i, j] + ave_i * (
```

Comment:
Could you please leave as it was? I find it really challenging to read in the new representation with more lines.

This is a Horner method mapped to code and this was the best method I found.
I appreciate the effort anyway though.